src/stac_utils/google.py

```diff
@@ -455,6 +495,73 @@ def send_data_to_sheets(
     return response
 
 
+def get_dataframe_from_drive(
+    file_id: str, delimiter: str, header=1, client: Resource = None, **kwargs
+) -> pandas.DataFrame:
+    """
+    Get csv (UTF-8 encoding) file from Google Drive, process for upload into BQ, and convert to DataFrame.
+    Make sure to share the file to "anyone with the link" temporarily, while using this method.
```

Comment:
if you share the file with the email associated with the service account you shouldn't need to make it public, i think